device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsDeviceOperations.java

```diff
@@ -427,6 +427,11 @@ protected boolean onLinkRemoteOpen(String linkName)
      */
     abstract protected AmqpsConvertToProtonReturnValue convertToProton(Message message) throws TransportException;
 
+    protected MessageImpl buildSubscribeToDesiredPropertiesProtonMessage()
```

Comment:
Since this is a twin only concept, you don't need an abstract method for this

Reply:
amqpsDeviceOperationMap value is an AmqpDeviceOperation, so I think I need to have it in there right?

Reply:
That map has a mix of classes that extend AmqpDeviceOperation. You can cast the twin one as a AmqpsDeviceTwin instance so that you can access the method that is twin specific

Reply:
FINE ! I will cast it :)